ignite/base/events_driven.py

```diff
@@ -304,14 +308,30 @@ def __setattr__(self, attr: str, value: Any) -> None:
 
         super().__setattr__(attr, value)
 
-    def update_mapping(self, event_to_attr: Mapping[Any, str]) -> None:
-        """Looks for the attribute and check if not existed to add it.
+    def update_mapping(self, attr_to_events: Mapping[str, List["Events"]]) -> None:
+        """Maps each attribute to a list of the corresponding events in the same given order.
 
         Args:
-            event_to_attr: mapping consists of the events from :class:`~ignite.engine.events.Events`
-                or any other custom events added by :meth:`~ignite.base.events_driven.EventsDriven.register_events`.
+            attr_to_events: mapping consists of the attributes mapped to a list events from
+                :class:`~ignite.engine.events.Events` or any other custom events added
+                by :meth:`~ignite.base.events_driven.EventsDriven.register_events`.
+                New attributes will be added, existed ones will be updated.
+                Getting attribute values is done based on the first element in the list of the events.
         """
-        for k, v in event_to_attr.items():
-            attr_evnts = self._attr_to_events[v]
-            if k not in attr_evnts:
-                attr_evnts.append(k)
+        if (
+            not isinstance(attr_to_events, dict)
+            or not all(isinstance(key, str) for key in list(attr_to_events.keys()))
+            or not all(isinstance(value, list) for value in list(attr_to_events.values()))
+        ):
+            raise TypeError(
+                "'attr_to_events' must be a dictionary, it shoule map keys of type string to list of Events"
+            )
+
+        for k, v in attr_to_events.items():
```

Comment:
Why we can't use `dict.update` method ?

Reply:
Can you consider this question ?

Reply:
Yes, updating here will update the already existed keys, if the user missed up with them or something that will lead to update them, which is we decided not to do this
> I think we should not alter existing attributes in the map, otherwise the API is cumbersome.

Reply:
I'm very unsure about `update_mapping` API and what we would like there. I'd like to simplify everything make strictly necessary things